src/go/rpk/pkg/cli/cloud/login.go

```diff
@@ -202,17 +203,31 @@ func loginProfileFlow(ctx context.Context, fs afero.Fs, y *config.RpkYaml, auth
 		if err != nil {
 			return "", err
 		}
-		c = nameAndCs[idx].c
+		selected = nameAndCs[idx]
 	}
 
 	// We have a cluster selected, but the list response does not return
 	// all information we need. We need to now directly request this
 	// cluster's information.
-	c, err = cl.Cluster(ctx, c.ID)
-	if err != nil {
-		return "", fmt.Errorf("unable to get cluster %q information: %w", c.ID, err)
+	var (
+		requiresMTLS, requiresSASL bool
+		p                          config.RpkProfile
+	)
+	if selected.c != nil {
+		c := *selected.c
+		c, err = cl.Cluster(ctx, c.ID)
```

Comment:
I don't love this initialize -> pass -> reassign. Feels needlessly confusing, especially when c only lives in this scope and does not provide additional clarity through the name. Below may get error for reinitializing err given how deep we are in the chain but IMO is a little easier to follow. 

```suggestion
		c, err := cl.Cluster(ctx, *selected.c.ID)
```

Reply:
True, fixed both. Thanks.